mods/tnt/init.lua

```diff
@@ -160,22 +155,54 @@ local function add_effects(pos, radius)
 		maxsize = 16,
 		texture = "tnt_smoke.png",
 	})
+
+	-- we just dropped some items. Look at the items entities and pick
+	-- one of them to use as texture
+	local texture = "tnt_blast.png" --fallback texture
+	local objs = minetest.get_objects_inside_radius(pos, 2)
+	for _, obj in pairs(objs) do
+		if obj and obj:get_luaentity() then
+			local def = ItemStack(obj:get_luaentity().itemstring):get_definition()
+			if def.tiles then
+				texture = def.tiles[1]
+				break
+			end
+		end
+	end
```

Comment:
You know, I actually had that coded up before, not sure why I didn't keep that code.

It's actually really trivial to do this, so, totally doing that :)